pages/docs/tutorials/native/dynamic-libraries.md

```diff
@@ -0,0 +1,346 @@
+```kotlin
+package demo
+
+class DemoClass {
+  fun foo() : Long = 42
+}
+
+fun ints(b: Byte, s: Short, i: Int, l:Long) { }
+fun floats(f: Float, d: Double) { }
+
+fun strings(str: String) : String {
+  return "That is '$str' from C"
+}
+
+val globalString = "A global String"
+```
+
+We need to have a Kotlin/Native compiler on our machines. 
+You may have a look at the
+[A Basic Kotlin/Native Application](basic-kotlin-native-app.html#obtaining-the-compiler)
+tutorial for more information on performing this step.
+Let's assume we have a console, where `kotlinc` command is available. 
+
+Now we call the following command to compile the code into a dynamic library:
+```bash
+kotlinc lib.kt -produce dynamic -output demo
+```
+
+The `kotlinc` generates the following files, depending on the OS, 
+where you run the compiler:
+- macOS: `demo_api.h` and `libdemo.dylib`
+- Linux: `demo_api.h` and `libdemo.so`
+- Windows: `demo_api.h` and `demo.dll`
+
+Let's check the C API for our Kotlin code in the `demo_api.h` 
+
+## Generated Headers File
+
+In the `demo_api.h` (with Kotlin/Native v0.8.1) you'll find the following code. 
+We discuss the code in parts to understand it easier. 
+
+The very first part contains standard C/C++ header and footer:
+```c
+#ifndef KONAN_DEMO_H
+#define KONAN_DEMO_H
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/// HERE GOES THE REST OF THE GENERATED CODE
+
+#ifdef __cplusplus
+}  /* extern "C" */
+#endif
+#endif  /* KONAN_DEMO_H */
+```
+
+After the rituals in the `demo_api.h` we have the block with common type definitions:
+```c
+#ifdef __cplusplus
+typedef bool            demo_KBoolean;
+#else
+typedef _Bool           demo_KBoolean;
+#endif
+typedef char            demo_KByte;
+typedef unsigned short  demo_KChar;
+typedef short           demo_KShort;
+typedef int             demo_KInt;
+typedef long long       demo_KLong;
+typedef float           demo_KFloat;
+typedef double          demo_KDouble;
+typedef void*           demo_KNativePtr;
+struct demo_KType;
+``` 
+
+Kotlin uses `demo_` prefix from the library name to make sure
+the symbols will not clash with other symbols in your C codebase.
+
+The definitions part shows how Kotlin primitive types map into C primitive types. 
+We discussed the opposite mapping the tutorial [INSERT THE LINK TO THE TUTORIAL FROM ANOTHER PULL REQUEST], 
+which you may want to check out. 
+
+The next part of the `demo_api.h` file contains definition of types
+that are used in the library:
+
+```c
+struct demo_KType;
+typedef struct demo_KType demo_KType;
+
+typedef struct {
+  demo_KNativePtr pinned;
+} demo_kref_demo_DemoClazz;
+
+```
+
+The `typedef struct { .. } TYPE_NAME` syntax is used in C language to declare a structure. 
+You may want to check [the thread](https://stackoverflow.com/questions/1675351/typedef-struct-vs-struct-definitions)
+for an explanation of that pattern.
+
+We see from that definitions that Kotlin object `DemoObject` is mapped into
```

Comment:
As with Objective-C manual not sure if diving into details of generated C headers is a good idea.